### PR TITLE
Feature/core 914 add onpage navigation

### PR DESCRIPTION
**Pull request by karladler**

### CODE REVIEW

File: core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      behavior: 'smooth',
+      left: element.offsetLeft,
+      top: element.offsetTop,
+    });
+  }
+
+  /**
+   * get the reference to the scroll container.
+   */
+  setScrollContainer(): void {
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.ts]: get or set? I would chose a different naming. "set" implies some kind of setter for me.

File: core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    const element = document.getElementById(this.entries[0].id);
+
+    this.scrollContainer = element.closest(this.scrollContainerSelector);
+  }
+
+  /**
+   * Iterate over target elements and add there y-positions.
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.ts]:
```suggestion
   * Iterate over target elements and add their y-positions.
```

File: core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+</template>
+
+<script lang="ts">
+import Component from './onpage-navigation';
+
+export default Component;
+</script>
+
+<style lang="scss" scoped>
+  @import './onpage-navigation.scss'
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.vue]: own scss file for little code 😢

**Comment by datrinh** [comment]
See comments

File: core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.scss [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+@import '~@evan.network/ui/src/style/utils';
+
+.onpage-nav {
+  position: fixed;
+  top: 80px;
+  right: 80px;
+
+  max-width: 200px;
+
+  ul {
```
[the remark this answers is not in the document]
Comment (karladler) [reply core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.scss]: it's already semantic bc targeted by `.onpage-nav ul` 🤓

**Pull request merged**